enterprise/migrations/0233_create_vw_enterprise_customer_ecus_pecus.py

```diff
@@ -0,0 +1,56 @@
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ('enterprise', '0232_backfill_ecu_table_user_fk_data_migration'),
+    ]
+    operations = [
+        # Custom SQL to add view that unions EnterpriseCustomerUser and PendingEnterpriseCustomerUser records
+        # together to facilitate querying both simultaneously
+        migrations.RunSQL(
+            sql="""
+                CREATE VIEW view_enterprise_customer_support_users AS
+                    SELECT ecu.enterprise_customer_id,
+                    au.email AS user_email,
+                    FALSE AS is_pending,
+                    ecu.user_id,
+                    au.username,
+                    au.first_name,
+                    au.last_name,
+                    au.is_staff,
+                    au.is_active,
+                    au.date_joined,
+                    EXISTS(
+                        SELECT * FROM enterprise_systemwideenterpriseuserroleassignment AS era
+                        INNER JOIN enterprise_systemwideenterpriserole AS ra ON era.role_id = ra.id
+                        WHERE era.enterprise_customer_id = ecu.enterprise_customer_id
+                        AND era.user_id = au.id
+                        AND ra.name = 'enterprise_admin'
+                    ) AS is_admin
```

Comment:
I think you could move this to the primary join block, and then compute `is_admin` with a CASE statement - I think it will be more efficient (maybe):
```sql
...
CASE WHEN ra.name = 'enterprise_admin' THEN true ELSE false END AS is_admin
FROM enterprise_enterprisecustomeruser AS ecu
INNER JOIN auth_user AS au ON au.id = ecu.user_fk
INNER JOIN enterprise_systemwideenterpriseuserroleassignment AS era ON era.enterprise_customer_id = ecu.enterprise_customer_id AND era.user_id = au.id
INNER JOIN enterprise_systemwideenterpriserole AS ra ON era.role_id = ra.id
```

Reply:
It was drastically more efficient, as it turned out.  Querying our largest production customer with the current implementation took around 7 seconds, with this it got down to ~1 second!

Now the bad news:  if an ecu has more than one role assigned to them, duplicate results will be returned for each role they have.

In the end, I went with a common table expression solution that @bcitro suggested, which brings the query down to ~5 seconds.  If you can think of a way to make the join work without producing duplicates, do let me know!